synthesis/codon/codon_test.go

```diff
@@ -108,27 +84,13 @@ func TestOptimize(t *testing.T) {
 func TestOptimizeSameSeed(t *testing.T) {
 	var gfpTranslation = "MASKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKFICTTGKLPVPWPTLVTTFSYGVQCFSRYPDHMKRHDFFKSAMPEGYVQERTISFKDDGNYKTRAEVKFEGDTLVNRIELKGIDFKEDGNILGHKLEYNYNSHNVYITADKQKNGIKANFKIRHNIEDGSVQLADHYQQNTPIGDGPVLLPDNHYLSTQSALSKDPNEKRDHMVLLEFVTAAGITHGMDELYK*"
 	var sequence, _ = genbank.Read("../../data/puc19.gbk")
-	var codonTable = GetCodonTable(11)
```

Comment:
The vast majority of examples and tests were using this "look for CDS" loop, and this is what we needed to compute stats. By moving this to its own function (`extractCodingRegion`) things are DRYer and we can compute stats every time we update weights.

Reply:
We should perhaps have the "look for CDS" loop in the interface - it is pretty common to use as a user interested in this package.

Reply:
Is there an existing package for our internal DNA data structures (or reading genbank files)? I think a `GetFeature()` func should probably live in there but otherwise can add it to this interface

Reply:
True, it could live there (in `io/genbank`, though this is changing a bit with a big PR). The special part is this isn't just looking for CDSs: it would be looking for CDSs that aren't pseudogenes and that don' t have stop codons. This is still pretty useful though to have there

Reply:
There's no internal DNA data structures yet, we currently represent sequences as strings. I want to change this though, see #347

Reply:
Commenting on that